### PR TITLE
Feat/o create position

### PR DESCRIPTION
**Pull request by onakutin**
Form works, a little issue with the console trying to log the error line 41 even if the 'try' executed correctly.

### CODE REVIEW

File: app/Http/Controllers/Api/PositionController.php [hunk 1]
```diff
@@ -44,7 +44,24 @@ public function create()
...
+        $position->name = $request->name;
+        $position->description = $request->description;
+        $position->hiring = 1;
+        $position->start_date = $request->start_date;
+        $position->end_date = $request->end_date;
+        $position->save();
+
+        return
+            [
+                'message' => 'succes'
```
Comment (EyK-26) [inline app/Http/Controllers/Api/PositionController.php:63]: with the message, you can return id of the position created by user as well, something like this :
        return
            [
                'message' => 'succes', 'id' => $position->id,
            ];

File: resources/js/hiroapp/components/recruiter/CreatePosition.jsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
+import axios from "axios";
+import React, { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+const CreatePosition = () => {
+	const navigate = useNavigate;
```
Comment (EyK-26) [inline resources/js/hiroapp/components/recruiter/CreatePosition.jsx:6]: calling the hook was missing, as I am familiar with typos it took time for me to see as well :)
const navigate = useNavigate();

File: resources/js/hiroapp/components/recruiter/CreatePosition.jsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+				[ev.target.name]: ev.target.value,
+			};
+		});
+	};
+
+	const handleSubmit = async (ev) => {
+		ev.preventDefault();
+		try {
+			const response = await axios.post("/api/positions", values);
+			navigate("/");
```
Comment (EyK-26) [inline resources/js/hiroapp/components/recruiter/CreatePosition.jsx:39]: you can redirect user to the created position so it's more user friendly and you don't need to clean the form afterwards:
so you can use something like:
navigate("/positions/" + response.data.id);

File: app/Http/Controllers/Api/PositionController.php (hunk 1 above)
Comment (EyK-26) [reply app/Http/Controllers/Api/PositionController.php:63]: that will help you to useNavigate below and redirect

File: resources/js/hiroapp/components/recruiter/CreatePosition.jsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+		fetchPositions();
+	}, []);
+
+	return (
+		<>
+			<h2>Create new position</h2>
+
+			<form action="/" method="post" onSubmit={handleSubmit}>
+				<br />
+				<label>
```
Comment (EyK-26) [inline resources/js/hiroapp/components/recruiter/CreatePosition.jsx:55]: I think label is not covering the elements inside them. I checked online and I see that label is only one line element.

**Review by EyK-26 — CHANGES_REQUESTED** [review]
Hi Ondra, I have added some comments to the code here. I think you can find them in files changed

File: resources/js/hiroapp/components/recruiter/CreatePosition.jsx (hunk 4 above)
Comment (onakutin) [reply resources/js/hiroapp/components/recruiter/CreatePosition.jsx:55]: https://react.dev/reference/react-dom/components/input#providing-a-label-for-an-input
https://react.dev/reference/react-dom/components/select

File: resources/js/hiroapp/components/recruiter/CreatePosition.jsx (hunk 3 above)
Comment (onakutin) [reply resources/js/hiroapp/components/recruiter/CreatePosition.jsx:39]: Of course you are right

File: app/Http/Controllers/Api/PositionController.php (hunk 1 above)
Comment (onakutin) [reply app/Http/Controllers/Api/PositionController.php:63]: Ok, I ammended it accordingly. Thanks a lot.

**Review by EyK-26 — APPROVED** [review]
thanks for the corrections, approved

**Pull request merged**